AutoGraph/WebSocketClient.swift

```diff
@@ -298,6 +325,41 @@ extension WebSocketClient: WebSocketDelegate {
         }
     }
     
+    func processWebSocketTextEvent(text: String) throws {
+        guard let data = text.data(using: .utf8) else {
+            throw ResponseSerializerError.failedToConvertTextToData(text)
+        }
+        
+        let json = try JSONValue.decode(data)
+        guard case let .string(typeValue) = json["type"] else {
+            return
+        }
+        
+        let graphQLWSProtocol = GraphQLWSProtocol(rawValue: typeValue)
+        switch graphQLWSProtocol {
+        case .connectionAck:
+            self.isConnectionAck = true
```

Comment:
self is not a connection ack.

`receivedConnectionAck` sounds more appropriate.

even better is if we could include this in state. we should be `.connected` at this point right? if so could we change `case connected` to `case connected(receivedAck: Bool)` and then do
```swift
guard self.state = .connected(_) else {
    return
}
self.state = .connected(receivedAck: true)
```
?

Reply:
I don't know if I think that is a better. I don't think they should be tied together

Reply:
I think using that flag helps break apart the connected state from if we receive connection ack

Reply:
But they are tied together if you can only get the ack if they are connected. The more I think about it, the more this sounds like the right approach.